cv_next/app/actions/cvs/preview.ts

```diff
@@ -0,0 +1,58 @@
+"use server";
+
+import Definitions from "@/lib/definitions";
+import { getIdFromLink, getGoogleImageUrl } from "@/helpers/imageURLHelper";
+import SupabaseHelper from "@/server/api/supabaseHelper";
+import { compareHashes } from "@/helpers/blobHelper";
+import MyLogger from "@/server/base/logger";
+
+const blobDataMap = new Map<string, Blob>();
+
+export async function revalidatePreview(cvId: string) {
+  const id = getIdFromLink(cvId);
+  const fileName = id + ".png";
+  const response = await fetch(getGoogleImageUrl(cvId), {
+    next: { revalidate: Definitions.FETCH_WAIT_TIME },
+  });
+  const blob = await response.blob();
+  const isSimilar = await compareHashes(
+    blob,
+    blobDataMap.get(id || "") || new Blob()
+  );
+  if (!isSimilar) {
+    blobDataMap.delete(id || "");
+    blobDataMap.set(id || "", blob);
+
+    const { data, error } = await SupabaseHelper.getSupabaseInstance()
+      .storage.from("cvs")
+      .upload(fileName, blob, {
+        cacheControl: "3600",
+        upsert: true, // Replace if exists
+      });
+
+    if (error) {
+      MyLogger.logInfo("Upload error:", error);
```

Comment:
Same comment as in other PR about error logging.

Reply:
Avner started working with winaton in the cv page with supabase
I'll add it to this branch as well and replace all

Reply:
See comment on other PR

Reply:
Resolving on other PR so we can focus this here
I'll create a new branch from dev
As this is the last thread for this PR, should we wait for the logger, merge it, and fix on each branch or merge these and fix in the logger branch?

Reply:
Merge this branch and fix on the logger branch